includes/Listeners/WPMail.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace NewfoldLabs\WP\Module\Data\Listeners;
+
+/**
+ * Monitors wp_mail events.
+ */
+class WPMail extends Listener {
+
+	/**
+	 * Register wp_mail_* hooks for the subscriber.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		add_action( 'wp_mail_succeeded', array( $this, 'mail_succeeded' ), 10, 1 );
+	}
+
+	/**
+	 * Mail sent successfully.
+	 *
+	 * @param array $mail_data An array containing the email recipient(s), subject, message, headers, and attachments.
+	 * @return void
+	 */
+	public function mail_succeeded( $mail_data ) {
+		$this->push(
+			'wp_mail',
+			array(
+				'label_key' => 'subject',
+				'subject'   => $mail_data['subject'],
```

Comment:
Can we be more defensive and make sure that the `subject` array key exists before sending this event? Also, have you been able to verify that this is properly sent to Google Analytics locally?

Reply:
I did not add that extra check since the subject was a required field (https://developer.wordpress.org/reference/functions/wp_mail/). It immediately throws a fatal error when the subject is removed from the function's argument.

Haven't been able to filter out the subject in any way using the available filters either. I can add the check if you still think it would be necessary.

Reply:
@arunshenoy99 That should be fine, then.

To test with GA, you can set up a new personal GA account for testing and set the info in the Hiive `.env` file.

Reply:
Thanks, @wpscholar. I was able to set up Hiive locally with a personal GA4 account, and I am seeing the `wp_mail` event flow through. Can we please include this in our release today?

![image](https://github.com/newfold-labs/wp-module-data/assets/38878906/66f63b4e-b330-4ad5-b4de-9d2ec7d8071a)

![image](https://github.com/newfold-labs/wp-module-data/assets/38878906/f17371fb-85a3-45f4-8a6d-90b48cbe64d8)